libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -564,9 +564,9 @@ void AP_UAVCAN::SRV_push_servos()
 {
     WITH_SEMAPHORE(SRV_sem);
 
-    for (uint8_t i = 0; i < NUM_SERVO_CHANNELS; i++) {
+    for (uint8_t i = 0; i < UAVCAN_SRV_NUMBER; i++) {
```

Comment:
no that isn't correct, see
`https://github.com/ArduPilot/ardupilot/blob/02219ba3e540a3f835b0ad9547487db8c8177c32/libraries/SRV_Channel/SRV_Channel_aux.cpp#L89` the function will only return something different to SRV_Channel::k_none if we are under NUM_SERVO_CHANNELS.
So raising to UAVCAN_SRV_NUMBER is just a waste of time if UAVCAN_SRV_NUMBER  > NUM_SERVO_CHANNELS as we won't do anything more.
if UAVCAN_SRV_NUMBER   < NUM_SERVO_CHANNELS  then, we got another issue !